Storage/tests/System/SignedUrlTest.php

```diff
@@ -142,37 +188,61 @@ public function testSignedUploadSessionOrigin()
         $this->assertEquals(self::CONTENT, $obj->downloadAsString());
     }
 
-    public function testSignedUrlContentType()
+    /**
+     * @dataProvider signingVersion
+     */
+    public function testSignedUrlContentType($version)
     {
+        $contentType = 'image/jpg';
+        $disposition = 'attachment;filename="image.jpg"';
         $obj = $this->createFile(uniqid(self::TESTING_PREFIX) .'.txt');
 
         $obj->update([
             'contentType' => null
         ]);
 
-        $url = $obj->signedUrl(time()+2, [
-            'responseDisposition' => 'attachment;filename="image.jpg"',
-            'responseType' => 'image/jpg'
+        $url = $obj->signedUrl(time() + 2, [
+            'responseDisposition' => $disposition,
+            'responseType' => $contentType,
+            'version' => $version
         ]);
 
         $res = $this->guzzle->request('GET', $url);
 
-        $this->assertEquals('image/jpg', $res->getHeaderLine('Content-Type'));
-        $this->assertEquals('attachment;filename="image.jpg"', $res->getHeaderLine('Content-Disposition'));
+        $expectedContentType = $version === 'v2'
+            ? urlencode($contentType)
+            : $contentType;
+
+        $this->assertEquals(200, $res->getStatusCode());
+        $this->assertEquals($expectedContentType, $res->getHeaderLine('Content-Type'));
+        $this->assertEquals($disposition, $res->getHeaderLine('Content-Disposition'));
     }
 
-    public function testSignedUrlWithSaveAsName()
+    /**
+     * @dataProvider signingVersion
+     */
+    public function testSignedUrlWithSaveAsName($version)
```

Comment:
Is it possible to create a signed URL to list objects using the PHP implementation of signed URLs?

Reply:
I have not added it yet, but the modifications in the change should make it straightforward.